### DISCUSSION TITLE
Top N slowest queries from AppInsights

### DISCUSSION THREAD

**Discussion by PeerMohamedMydeenA** [Azure/data-api-builder#2096] [Q&A]
I'm looking for a way to build a report for top N slowest queries, slowest queries by entities, duration, result size etc. 

When I looked at the telemetry generated by DAB, I find the SQL queries generated by DAB from the graphql query in traces and no way to tie that to the requests to find the slowest queries. 

Is there any other way to capture the original grphql query and the duration  ?

**Comment by seantleonard** [comment]
Hi @PeerMohamedMydeenA , while we don't have that today, @jerrynixon is working on list of stats /perf counters to collect and which would be most useful.

**Comment by yorek** [comment]
If you are using MSSQL, you can use regular monitoring tools like [Performance Insights](https://learn.microsoft.com/en-us/azure/azure-sql/database/query-performance-insight-use?view=azuresql) or using [performance system views](https://learn.microsoft.com/en-us/troubleshoot/sql/database-engine/performance/troubleshoot-slow-running-queries). Would be that a good alternative for now?

**Reply by PeerMohamedMydeenA** [reply]
@yorek, this is certainly a good alternate in case of MSSQL. We are also looking into using DAB with CosmosDB as a variation in platform offering. Having these metrics unified onto AppInsights would help us building Reports out of Log Analytics.
